synnax/pkg/distribution/channel/channel_group.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+//
+// channelGroup implements tools for constructing a set of channels. The set of
+// channels is constructed as a sorted array of channelRanges. Each channelRange
+// consists of the local key that the range starts at and the size of the
+// channelRange (how many consecutive keys it is).
+
+package channel
+
+import (
+	"slices"
+	"sort"
+)
+
+// channelRange is the basic struct for ChannelGroup, and is defined as a set of
+// consecutive channel keys.
+type channelRange struct {

```

Comment:
There's no point in starting the name of a struct with the package name unless the struct has the name of the package itself. This should just be called `Range`, or, even better, `KeyRange`